### DISCUSSION TITLE
Networking between existing apps and jail on TrueNAS Scale Dragonfish-24.04.1.1

### DISCUSSION THREAD

**Discussion by SonicJoe** [Jip-Hop/jailmaker#202] [Q&A] [answered]
I'm not sure if this is a known thing, something I'm doing wrong, or what. I am in the process of moving my apps over to jailmaker. I used the docker template to build my jail and following the linked YouTube video I setup Dockge. So far, so good.

Now, I have avoided using Truecharts for apps for most things, but I do have 2 apps that TrueNAS native charts didn't have (Flaresolverr and Unpackerr). I figured Flaresolverr would be a good "test" of my jail (no environment variables or data sets to mount, a very basic app) so I deployed that using Dockge and its working. If I go to http://192.168.20.100:8191 (my jail's IP) I get a response from Flaresolverr and I see in the logs that it logged the connection. However, I then tried to point my Prowlarr instance (running the TrueNAS native Prowlarr) at the new IP and it fails to connect. No connection is logged. I've run the following tests:

1) From Prowlarr shell, ping docker jail: Fails
2) From Prowlarr shell, ping network GW: Success (they're on the same network, but I figured the GW was a good neutral IP to test with)
3) From Prowlarr shell, ping TrueNAS IP: Success
4) From docker jail shell, ping TrueNAS IP: Success
5) From docker jail shell, ping network GW: Success
6) From computer on LAN subnet, go to http://192.168.20.100:8191 (Flaresolverr on docker jail), get response
7) From computer on LAN subnet, go to http://192.168.20.50:8191 (Flaresolverr Truecharts app), get response
8) From TrueNAS shell, curl http://192.168.20.100:8191, get response
9) From docker jail shell, curl http://192.168.20.50:8191, get response

At this point, I am lost. All my network checks check out, but for some reason Prowlarr cannot reach this one IP. My guess is something about the TrueNAS apps network, but why just the jail IP?

Anyway, is this something anyone else has run into and is there a way to make this work? I was trying to avoid a full lift and shift to jailmaker, preferring to migrate one app at a time, but I guess if I have to tear it all down that's what I'll do.

**Comment by Lockszmith-GH** [comment]
I would recommend you share more details about your setup (per #135 - How to ask for help? specifically the [Troubleshooting and Collecting infromation](135#discussioncomment-9215396) section)

We cannot make assumptions of your setup from your description. Are you using a bridge or macvlan? How is your K3s cluster configured? Are there any static routes setup on your TrueNAS networking setup? Are there any specific interfaces assigned.

There are probably other questions as well, gather as much information as you can, and maybe we can figure out together where to look next for the solution

**Reply by SonicJoe** [reply]
Sure, I can give you whatever you want. I was hoping maybe someone had just seen it before and would either be like "yeah, that won't work" or "oh yeah, you have to set this in your jail for that to work".

In answer to your questions:

1. Are you using a bridge or macvlan? Sorry, I did mean to include this. I am using bridge.
2. How is your K3s cluster configured? Since TrueNAS deploys all of that, I'm not sure I can answer that. I don't remember having to do anything other than to pick a dataset for the Applications and then adding apps. Certainly if there is any customization options, I didn't touch anything and it is whatever TrueNAS defaults to. If there is more info you need and you can tell me what commands to run and where to run them I'll get whatever relevant info you want.
3. Are there any static routes setup on your TrueNAS networking setup? Yes, I have a default route added using the "server network" gateway. 0.0.0.0/0 to 192.168.20.1. I did this because I do have an interface on my NAS connected to my LAN so that my backup and SMB traffic wouldn't have to cross my firewall.
4. Are there any specific interfaces assigned? I'm not sure what you mean by this. What specific interfaces and where? On TrueNAS, in the jail, in the apps, etc? 

<details><summary>jlmkr edit docker</summary>
startup=1
gpu_passthrough_intel=0
gpu_passthrough_nvidia=0
\# Turning off seccomp filtering improves performance at the expense of security
seccomp=1

\# Use macvlan networking to provide an isolated network namespace,
\# so docker can manage firewall rules
\# Alternatively use --network-macvlan=eno1 instead of --network-bridge
\# Ensure to change eno1/br1 to the interface name you want to use
\# You may want to add additional options here, e.g. bind mounts
systemd_nspawn_user_args=--network-bridge=br0
        --resolv-conf=bind-host
        --system-call-filter='add_key keyctl bpf'
        --bind='/mnt/Application/docker/data:/mnt/data'
        --bind='/mnt/Application/docker/stacks:/opt/stacks'

\# Script to run on the HOST before starting the jail
\# Load kernel module and config kernel settings required for docker
pre_start_hook=\#!/usr/bin/bash
        set -euo pipefail
        echo 'PRE_START_HOOK'
        echo 1 > /proc/sys/net/ipv4/ip_forward
        modprobe br_netfilter
        echo 1 > /proc/sys/net/bridge/bridge-nf-call-iptables
        echo 1 > /proc/sys/net/bridge/bridge-nf-call-ip6tables

\# Only used while creating the jail
distro=debian
release=bookworm

\# Install docker inside the jail:
\# https://docs.docker.com/engine/install/debian/\#install-using-the-repository
\# Will also install the NVIDIA Container Toolkit if gpu_passthrough_nvidia=1 during initial setup
\# https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html
initial_setup=\#!/usr/bin/bash
        set -euo pipefail

        apt-get update && apt-get -y install ca-certificates curl
        install -m 0755 -d /etc/apt/keyrings
        curl -fsSL https://download.docker.com/linux/debian/gpg -o /etc/apt/keyrings/docker.asc
        chmod a+r /etc/apt/keyrings/docker.asc

        echo \
        "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/debian \
        $(. /etc/os-release && echo "$VERSION_CODENAME") stable" | \
        tee /etc/apt/sources.list.d/docker.list > /dev/null

        apt-get update
        apt-get install -y docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin

        \# The /usr/bin/nvidia-smi will be present when gpu_passthrough_nvidia=1
        if [ -f /usr/bin/nvidia-smi ]; then
        curl -fsSL https://nvidia.github.io/libnvidia-container/gpgkey -o /etc/apt/keyrings/nvidia.asc
        chmod a+r /etc/apt/keyrings/nvidia.asc
        curl -s -L https://nvidia.github.io/libnvidia-container/stable/deb/nvidia-container-toolkit.list | \
        sed 's\#deb https://\#deb [signed-by=/etc/apt/keyrings/nvidia.asc] https://\#g' | \
        tee /etc/apt/sources.list.d/nvidia-container-toolkit.list
        apt-get update
        apt-get install -y nvidia-container-toolkit

        nvidia-ctk runtime configure --runtime=docker
        systemctl restart docker
        fi

        docker info

\# You generally will not need to change the options below
systemd_run_default_args=--property=KillMode=mixed
        --property=Type=notify
        --property=RestartForceExitStatus=133
        --property=SuccessExitStatus=133
        --property=Delegate=yes
        --property=TasksMax=infinity
        --collect
        --setenv=SYSTEMD_NSPAWN_LOCK=0

systemd_nspawn_default_args=--keep-unit
        --quiet
        --boot
        --bind-ro=/sys/module
        --inaccessible=/sys/module/apparmor
</details>

**Reply by Lockszmith-GH** [reply]
While I understand the sentiment - about asking whether someone encountered a similar issue - I did so as well across the years, these questions tend to be an [XYproblem](https://xyproblem.info) and sharing details is usually the fastest route to get to root causes.

* for 2. run the command below - the output is equivalent to what you'll see in Apps -> Advance Settings.
```
cli -c 'app kubernetes config'
```
* for 3. the following command's output might be helpful: (note, these include all of your IP configuration, so sanitize as needs, just mention what is sanitized)
```bash
cli -c 'network general summary'
cli -c 'network configuration config'
```
* 4. will be answered by 2. above.

**Reply by Lockszmith-GH** [reply]
as for 3. - is the NAS IP range and your LAN IP range different?
If not, then the route is unnecessary.

Also, what's the network setup within the jail?
And what networking setup are you using for docker?

**Reply by SonicJoe** [reply]
Thanks. I totally understand as well. I work in IT so I totally get the "user didn't provide enough info" thing, but, well, sometimes its also good to know what you don't know and just wait until someone asks you for things then posting irrelevant info like crazy. I understand I am taking this from a "network engineer" perspective, which may not be a valid approach. If you are unclear about what I am doing and/or am trying to accomplish then by all means ask and I will do my best to clarify. I appreciate the effort.

cli -c 'app kubernetes config'

```
+--------------------+-----------------------------+
|                 id | 1                           |
|               pool | Application                 |
|       cluster_cidr | 172.16.0.0/16               |
|       service_cidr | 172.17.0.0/16               |
|     cluster_dns_ip | 172.17.0.10                 |
| route_v4_interface | br0                         |
|   route_v4_gateway | 192.168.20.1                |
| route_v6_interface | <null>                      |
|   route_v6_gateway | <null>                      |
|            node_ip | 192.168.20.50               |
|     configure_gpus | true                        |
|          servicelb | true                        |
|   passthrough_mode | false                       |
|     metrics_server | false                       |
|            dataset | Application/ix-applications |
+--------------------+-----------------------------+

```

cli -c 'network general summary'

```
+----------------+--------------+
|            ips | <dict>       |
| default_routes | 192.168.20.1 |
|    nameservers | 192.168.20.1 |
+----------------+--------------+

```
cli -c 'network configuration config'

```
+----------------------+---------------+
|                   id | 1             |
|             hostname | NJ-NAS-01     |
|               domain | <removed> |
|          ipv4gateway | 192.168.20.1  |
|          ipv6gateway |               |
|          nameserver1 | 192.168.20.1  |
|          nameserver2 |               |
|          nameserver3 |               |
|            httpproxy |               |
|                hosts | <empty list>  |
|              domains | <empty list>  |
| service_announcement | <dict>        |
|             activity | <dict>        |
|       hostname_local | NJ-NAS-01     |
|                state | <dict>        |
+----------------------+---------------+
```

All I removed was my domain name since since it is public so I can issue certificates for things.

My NAS's main interface is on what I call my "server network" where I have my NAS, and a Proxmox host with a few VMs/LXCs running. So I bound Samba to a separate interface and connected that to my LAN to avoid traversing the firewall when running my backups or accessing SMB shares. I think I was seeing some random traffic going out the wrong interface and so I added the static default route and that had stabilized things. I don't entirely remember as I did that probably a year ago or more. You are correct that I shouldn't need it since the system default gateway is the server network gateway any way. I removed it for now and I am not seeing any issues so I will leave it for now if it simplifies things while we troubleshoot. However, it didn't change the issue status.

The only change I did inside the jail is the recommended network changes from the networking part of the Read Me. Here is what's inside my 80-container-host0.network file (I commented out the original network section instead of overwriting it entirely, but that should not make a difference):

```
# SPDX-License-Identifier: MIT-0
#
# This config file is installed as part of systemd.
# It may be freely copied and edited (following the MIT No Attribution license).
#
# To make local modifications, one of the following methods may be used:
# 1. add a drop-in file that extends this file by creating the
#    /etc/systemd/network/80-container-host0.network.d/ directory and creating a
#    new .conf file there.
# 2. copy this file into /etc/systemd/network or one of the other paths checked
#    by systemd-networkd and edit it there.
# This file should not be edited in place, because it'll be overwritten on upgrades.

# This network file matches the container-side of the virtual Ethernet link
# created by systemd-nspawn's --network-veth switch. See systemd-nspawn(1) for
# details.

[Match]
Virtualization=container
Name=host0

#[Network]
#DHCP=yes
#LinkLocalAddressing=ipv6
#LLDP=yes
#EmitLLDP=customer-bridge
[Network]
DHCP=false
Address=192.168.20.100/24
Gateway=192.168.20.1
LinkLocalAddressing=no
LLDP=yes
EmitLLDP=customer-bridge


[DHCP]
UseTimezone=yes


```

As for Docker, I'm a noob so bare with me. I haven't done any customization to the networking beyond the base install using the template provided. This was my first app to attempt to deploy on my own without a tutorial to back me up. That being said, does this help? Its the output of docker network inspect flaresolverr_default

```
[
    {
        "Name": "flaresolverr_default",
        "Id": "9fe6cebffb550a020a958e5ef9340c8b839a2c56f13bc660a5bded6b11d4918e",
        "Created": "2024-06-16T21:32:46.510131688-04:00",
        "Scope": "local",
        "Driver": "bridge",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": [
                {
                    "Subnet": "172.19.0.0/16",
                    "Gateway": "172.19.0.1"
                }
            ]
        },
        "Internal": false,
        "Attachable": false,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": {
            "e04fb5913f4b8b28ad4ffa8b94cc99f6cc1d7aa94e115f9663a2cd83f6346d62": {
                "Name": "flaresolverr",
                "EndpointID": "aabc0e135496b8b91a72a180e7b9efde111c101d85f4ae669fea9e1e27cd5979",
                "MacAddress": "02:42:ac:13:00:02",
                "IPv4Address": "172.19.0.2/16",
                "IPv6Address": ""
            }
        },
        "Options": {},
        "Labels": {
            "com.docker.compose.network": "default",
            "com.docker.compose.project": "flaresolverr",
            "com.docker.compose.version": "2.21.0"
        }
    }
]

```

I should note again that my flaresolverr container is accessible from literally every other device/VM that I have tried from on my network, so I don't believe this is a "network" issue, at least not in an external routing/firewall sense. It is only when trying to access from a TrueNAS app that I have the issue.

**Reply by Lockszmith-GH** [reply]
I'm the same with you, ex Network IT guy with software developmeent and database development background - can't call myself engineer, because everything was self-taught or OJT, so there are gaps - but have enough experience under my belt.
I've been using docker since 2016, still can't call myself an expert, because it's mostly a hobby, other that one year ago, I didn't actually deploy a lot of production envrionments based on docker (something that is usually focused on development anyway), and my kubernetes experience is from TrueNAS SCALE (via TrueCharts) - until this just blew up.

So don't feel bad, you're in good company (I hope 🤭)

Anyway, when I think of processes, flows etc, I think visually, and so I drew what I understood from your setup and findings.
Is the diagram below a correct representation of your setup?

How is the Prowlarr app's networking defined in TrueNAS ?
I would also suggest is to find out what the routing looks like inside the Prowlarr pod, maybe that will give us the hint on why the connection isn't made.

I can tell you that I run a DNS server inside a jail, and my TrueNAS Chart apps reach the DNS server, but I'm not running any TrueNAS SCALE native apps.

```mermaid
flowchart
    direction TB
    subgraph gw["Gateway"]
        subgraph gw_c["Settings
            ip: 192.168.20.1
        "]
        end
    end

    url(["Click here to open diagram in live editor"])
    subgraph c["Computer on LAN"]
        subgraph c_c["Settings
            subnet: 192.168.20.0/24 ?
        "]
        end
        c_curl_jdf_p(["curl http://192.168.20.100:8191/"])
    end

    subgraph t["TrueNAS SCALE"]
        direction TB

        subgraph t_s["shell"]
            t_curl_jdf_p(["curl http://192.168.20.100:8191/"])
        end
        subgraph t_c["Settings
            hostname:        NJ-NAS-01
            domain:          &lt;removed&gt;
            ipv4gateway:     192.168.20.1
            nameserver1:     192.168.20.1
            hostname_local:  NJ-NAS-01
        "]
        end
        subgraph k["TrueNAS Apps (k3s)"]
            direction TB
            subgraph k_c["settings:
            id:                 1
            pool:               Application
            cluster_cidr:       172.16.0.0/16
            service_cidr:       172.17.0.0/16
            cluster_dns_ip:     172.17.0.10
            route_v4_interface: br0
            route_v4_gateway:   192.168.20.1
            route_v6_interface: null
            route_v6_gateway:   null
            node_ip:            192.168.20.50
            configure_gpus:     true
            servicelb:          true
            passthrough_mode:   false
            metrics_server:     false
            dataset:            Application/ix-applications
            "]
                
            end
            subgraph kp["Prowlarr"]
                direction TB
                subgraph kp_c["Settings:
                Type: TrueNAS SCALE 'native' (not TrueCharts)
                "]
                end
                subgraph kp_s["shell"]
                    kp_p_gw(["ping gateway"])
                    kp_p_t(["ping TrueNAS"])
                    kp_p_j(["ping jail"])
                end
            end
        end
        subgraph j["jail (docker)"]
            direction TB
            subgraph j_c["Settings:
                Jail name: docker
                Static IP: 192.168.20.100/24
                Gateway: 192.168.20.1
            "]
            end
            subgraph j_s["shell"]
                j_p_gw(["ping gateway"])
                j_p_t(["ping TrueNAS"])
                j_curl_jdf_p(["curl http://192.168.20.100:8191/"])
            end
            subgraph jd["docker"]
                direction TB
                subgraph jdf["container: Flaresolverr"]
                    direction TB
                    subgraph jdf_c["Settings:
                        networking: bridge mode (default)
                    "]
                    end
                    jdf_p((("port 8191")))
                end
            end
            subgraph j_n["Network interfaces"]
                br0_j((("br0")))
            end
        end
        subgraph t_n["Network interfaces"]
            eno0((("eno0")))
            br0((("br0")))
            eno0 -.->|Assign| br0
        end
    end

    br0                -...->|Assign | br0_j
    kp_p_t               --->|Success| br0
    j_p_t                --->|Success| br0
    kp_p_gw             ---->|Success| gw
    j_p_gw       ----------->|Success| gw
    kp_p_j        ---------->|Fail   | j
    c_curl_jdf_p ----------->|Success| jdf_p
    t_curl_jdf_p    -------->|Success| jdf_p
    j_curl_jdf_p      ------>|Success| jdf_p

    url o--o c
    click url "https://mermaid.live/edit#pako:eNqlV1tv2zYU_iuEBrQ2ENlWlqWtCnTwsrVYUQQFnLeqEBiJ1sUyKZCUvaDpf9-haN0oKk5bPknUp3POd67kNydiMXF8Z1uwY5RiLgOKYMUZJ5HMGEV3f-kdUd0nHJcpSo5fAucDluSIHwLnq_5qIMIIMBsiZUYT0SHUykofeW8uF97168XlauF1XwfCCI31S_2gHytezEDuTZFFO5QSTpBkiJWEgrkYVO9RRlGRHQgicSYZB4Fzw3hl1w3bl5UkHAG7T-tbO4foCQoAokQOaKyWl1foz3Nc1ALBwCPM421YKjbqDaVSlv5y2ffLauW_9t54y45EzxOtmRIk3PGK3K43aHOz_vTPQPEwihaWMhQgQKSkKAY_qiV_2tAR556-aa-mTEiK98RvNm4_ukDLXXlDXMz2OKN-t_GikG852bMDiV8k8q2ZboerRCer_sWee2op5YLwA-HeOWhja1iwCBe-1dbpFGjdsetFb12WAs12v4v5KBLjYhxLqh0rTo71DR_EPjKXQahkrDBBYBFUGlaKh-CoqATUTxhlMW9-8l4pXy1UJXjXhpHg0ywiY_grK7yRHlMRqmYxgHurIZgzqOTwcBVmFP7Z4gjS555PgXqJMB3bE_q6L5JWUCB2VE_mGEWhubYkGuqd5j8MQyNGt1lScRImZSX0XxLyw-rP4r4ndowqsRAyBTOTNNyDGQq8xYUwYHsieRaJUCe-lmiBxVhioXqePT-W2X8u7l6Nyh4ldL052BnUxzCzS0jsz5wdC8y5VdJ0eRiCBs3HH0PvHkpw06CdopcUGB3ISzSjTNbfbtSUFHMLI5txI2KmTdMNuFkAKsPkqHpwCZajpJ298yd-kC3-ROgMPm_xOc4KO3hEZrBh73E5iFUS0Sxm0Y7wn2pv-bnQfVQa9PTQasaQjYRIRujfz8Pzx0qN7jH6Q1PW041iRGQ6ifMzYc5_IML584Ob_9IIP0MpBoEnX_9CVYJtyjBGJQx11YDeQ50TwQroRnyyJJ4Wbio4lzxttybyyPgOUGqIZHFCkOqckLlki6tCTpTPpJXW2q_jUgdkNoMYMi6R8n3gzOc_XHFGjlGgeas5oHZ4Cat5MCOh4pUFalqOdT-jsOVz9RHKVrUq9WDRBRY8aQlbIXfhvntcC5El9HE43233BACYfN3FohOBHjV_jdbd0sS7gN5UEfARPY25BTqFPbVtE9rHJsdObIt0u2WB6l6NRth3j-9VC0TA7cSrf9GYEFp_0-j-ab8v2Y7ODXSDH6PbmxtirstQdLKtvsKp3cBR_UhAQ9oTDgf7eKEucEt1gfutxDvmQ-ACBzW3PsuFr7nrofC-wHTnXDgnSXCr_abUBY5MyZ4Ejg-PMea7wAnod8DhSrLNA40cXx2fLpz6sOT49fHnwqlKOPSQv7UiDfn-P6B2CJw" "Click to open diagram in editor" _blank
```

**Reply by SonicJoe** [reply, marked as answer]
Wow, that looks pretty good to me at first glance. I appreciate the hard work and the help, but I may have actually literally solved my issue as you were posting this.

I noticed in the TrueNAS GUI that I had a new interface (I assume this is created by the "jlmkr create" command) called "vb-docker". I bound this to my br0 interface in the TrueNAS GUI, and now everything can talk to each other just fine. Its interesting that that would break comms between K3s and the jail, but not the host and the jail, but it seems to have done the trick. I double-checked the documents and I don't see anything about needing to do this, so I don't know if this is a "bug" or just something finicky about my system.

Of course, if anyone sees a reason why doing this would be a bad idea, please let me know. Otherwise, I believe this is the solution.

**Reply by SonicJoe** [reply]
I'm marking that as the solution, in case anyone else runs into the same thing. Now that I have added the vb-docker interface as a member interface under my br0 interface in the TrueNAS GUI, this is definitely now working as expected and has been stable for 24 hours. Appreciate the help given.

![image](https://github.com/Jip-Hop/jailmaker/assets/4595977/8d4ba65d-d88a-4b67-a1af-b3d7889861f4)

**Reply by Lockszmith-GH** [reply]
That IS strange, because cb-docker is supposed to be bound to br0 to begin with.
I'm glad things have been resolved though.

**Reply by Matt-Of-Doom** [reply]
I've made an account to write a reply for I'm having this exact same issue, but worse.

When I try to ping from my jail or from one of the apps, I am unable too reach either of them.

I noticed the vb-docker as well after making a bridge (in the hopes that would fix my issues), but alas my luck is not the same. after doing so I still cannot commuincate across the docker images and the truenas apps. 

I've read through what you've written 3 times over, and my bridge file is set up the same way, just different IP addresses. 

Would it be possible to get a picture of your bridge setup fully? I see your jails ip's are ending in 100 and the apps in 50 so wondering if it's also my docker image setup of my jail or docge itself. Any help here would be appreciated!

**Reply by Lockszmith-GH** [reply]
Hi @Matt-Of-Doom, it look like @SonicJoe answered you [below](https://github.com/Jip-Hop/jailmaker/discussions/202#discussioncomment-10045441). 

If that's not the answer, please Please open a [new discussion](https://github.com/Jip-Hop/jailmaker/discussions/new?category=q-a). You can reference this discussion by typing `#` followed by the discussion number (or just search the title and the editor will fill it up) (in this particular case it's `#202`.

**Comment by SonicJoe** [comment]
So your TrueNAS box should be using the bridge interface as its IP, and then you join vb-docker to the bridge. However, I was actually going to update this since I had an issue upon reboot. I would not recommend this as a long term solution, as it causes issues bringing the network up on reboot. Its fine for a short duration as you migrate things, and as long as you have access to the console in case your system reboots. Anyway, my networking looks like this:

![image](https://github.com/user-attachments/assets/830a235d-86cd-4155-89a0-0008ab587bd0)

enp4s0 is a standalone interface going to my "LAN" network
I moved enp7s0 to enp5s0 when I was troubleshooting the aforementioned issue, so enps05 is now joined to br0 instead of enp7s0 as previously.

The bridge settings would look like this after you add vb-docker:

![image](https://github.com/user-attachments/assets/88f04246-2ee8-4abd-970e-84d2588f805f)

Just, remove vb-docker from br0 before rebooting for any reason, as the bridge interface won't come up if its there on boot. Or at least, it wouldn't for me.

**Reply by Matt-Of-Doom** [reply]
Interesting, Thank you for the pictures! I had mine set up in a similiar way, but seems like there might be some key differences. good to know about removing vb-docker if going for a reboot.

To clairfy you have enp4s0 which is your LAN is where you TrueNas box sits

Both enp4s0 and enp5s0 are physical interfaces on that device

br0 connects enp5s0 and the vb-docker together

I also see your bridge IP address range and your enp4s0 IP address range are different. so the question is 

Did you intnetionally seperate them into 2 different IP address and if so how are you getting them to talk to one another? If this is out of the scope of TrueNas that is fine but seems like your jails are in 1 network and the truenas box is on another. Is this something that can be set up in scale?

**Reply by SonicJoe** [reply]
> To clairfy you have enp4s0 which is your LAN is where you TrueNas box sits

No, my TrueNAS main interface sits on a network I call my "server network", which is the one enp5s0 is connected to.

> Both enp4s0 and enp5s0 are physical interfaces on that device

Correct

> br0 connects enp5s0 and the vb-docker together

Yes, and enp5s0 connects everything physically to the rest of the network

> I also see your bridge IP address range and your enp4s0 IP address range are different. so the question is
> 
> Did you intnetionally seperate them into 2 different IP address and if so how are you getting them to talk to one another?

Yes, I believe I mentioned it earlier when working with Lockszmith-GH. enp4s0 is on my LAN which is the network where all my personal devices live. Windows PCs, mobile phones, tablets, etc. The bridge interface is on my "server network" where all my services live. Really just a virtual wireless controller and a few other VMs I have for playing/learning with, nothing fancy. I set it up that way so when my LAN devices are accessing my SMB share the firewall doesn't have to process all of that traffic. The firewall handles routing between the two networks for other services.

>  If this is out of the scope of TrueNas that is fine but seems like your jails are in 1 network and the truenas box is on another. Is this something that can be set up in scale?

My jail is on the same network as my TrueNAS box. If you had more than one network, I believe there is a way to make the jail bound to more than one network. I know I saw something in the jailmaker documentation, but its not my use case so I can't help there. For binding to a single network, you just set that first line in the arguments per the template:

systemd_nspawn_user_args=**--network-bridge=br0**

That tells systemd-nspawn to bind the container to the bridge interface with label br0. There are other options you can use, if you look through the documentation.
